src/infrastructure/fragmenter.rs

```diff
@@ -60,30 +61,21 @@ impl Fragmentation {
     }
 
     /// Split the given payload into fragments and write those fragments to the passed packet data.
-    pub fn spit_into_fragments(
-        payload: &[u8],
-        acked_header: AckedPacketHeader,
-        packet_data: &mut PacketData,
+    pub fn spit_into_fragments<'a>(
+        payload: &'a [u8],
         config: &Config,
-    ) -> Result<()> {
+    ) -> Result<Vec<&'a [u8]>> {
+        let mut fragments = Vec::new();
+
         let payload_length = payload.len() as u16;
         let num_fragments =
-            Fragmentation::total_fragments_needed(payload_length, config.fragment_size) as u8; /* safe cast max fragments is u8 */
+            Fragmentation::fragments_needed(payload_length, config.fragment_size) as u8; /* safe cast max fragments is u8 */
```

Comment:
Can we put the comment above the line just to keep the style consistent? I'm also not totally sure what this comment means.

```suggestion
            // Safe cast max fragments is u8
            Fragmentation::fragments_needed(payload_length, config.fragment_size) as u8; 
```